crates/edr_provider/src/data.rs

```diff
@@ -2845,4 +3394,292 @@ mod tests {
+                max_fee_per_gas: Some(U256::ZERO),
+                ..default_call.clone()
+            },
+        )?;
+
+        assert_decoded_output(result.execution_result)?;
+
+        // Should accept pre-EIP-1559 gas semantics when running in the context of a
+        // pre-EIP-1559 block
+        let result = call_hello_world_contract(
+            &fixture.provider_data,
+            BlockSpec::Number(EIP_1559_ACTIVATION_BLOCK - 1),
+            CallRequest {
+                gas_price: Some(U256::ZERO),
+                ..default_call.clone()
+            },
+        )?;
+
+        assert_decoded_output(result.execution_result)?;
+
+        // Should throw when given post-EIP-1559 gas semantics and when running in the
+        // context of a pre-EIP-1559 block
+        let result = call_hello_world_contract(
+            &fixture.provider_data,
+            BlockSpec::Number(EIP_1559_ACTIVATION_BLOCK - 1),
+            CallRequest {
+                max_fee_per_gas: Some(U256::ZERO),
+                ..default_call.clone()
+            },
+        );
+
+        assert!(matches!(
+            result,
+            Err(ProviderError::RunTransaction(
+                TransactionError::Eip1559Unsupported
+            ))
+        ));
+
+        // Should accept pre-EIP-1559 gas semantics when running in the context of a
+        // post-EIP-1559 block
+        let result = call_hello_world_contract(
+            &fixture.provider_data,
+            BlockSpec::Number(EIP_1559_ACTIVATION_BLOCK),
+            CallRequest {
+                gas_price: Some(U256::ZERO),
+                ..default_call.clone()
+            },
+        )?;
+
+        assert_decoded_output(result.execution_result)?;
+
+        // Should support a historical call in the context of a block added via
+        // `mine_and_commit_blocks`
+        let previous_block_number = fixture.provider_data.last_block_number();
+
+        fixture.provider_data.mine_and_commit_blocks(100, 1)?;
+
+        let result = call_hello_world_contract(
+            &fixture.provider_data,
+            BlockSpec::Number(previous_block_number + 50),
+            CallRequest {
+                max_fee_per_gas: Some(U256::ZERO),
+                ..default_call
+            },
+        )?;
+
+        assert_decoded_output(result.execution_result)?;
+
+        Ok(())
+    }
+
+    macro_rules! impl_full_block_tests {
+        ($(
+            $name:ident => {
+                block_number: $block_number:expr,
+                chain_id: $chain_id:expr,
+                url: $url:expr,
+            },
+        )+) => {
+            $(
+                paste::item! {
+                    #[test]
+                    fn [<full_block_ $name>]() -> anyhow::Result<()> {
+                        let url = $url;
+                        run_full_block(url, $block_number, $chain_id)
+                    }
+                }
+            )+
+        }
+    }
+
+    impl_full_block_tests! {
```

Comment:
These tests result in way longer test times for Rust:

```
...
test data::tests::full_block_sepolia_shanghai ... ok
test data::tests::full_block_mainnet_istanbul ... ok
test data::tests::full_block_mainnet_byzantium ... ok
test data::tests::full_block_mainnet_constantinople ... ok
test data::tests::full_block_mainnet_muir_glacier ... ok
test data::tests::full_block_goerli_merge has been running for over 60 seconds
test data::tests::full_block_mainnet_shanghai has been running for over 60 seconds
test data::tests::full_block_goerli_merge ... ok
test data::tests::full_block_mainnet_shanghai ... ok

test result: ok. 54 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 201.13s
```

Should we consider excluding them from the pre-commit hook script?

Reply:
I'd keep them to keep the slow performance visible. We can reconsider if it turns it's low prio.